rlwe/encryptor.go

```diff
@@ -0,0 +1,509 @@
+package rlwe
+
+import (
+	"github.com/ldsec/lattigo/v2/ring"
+	"github.com/ldsec/lattigo/v2/utils"
+)
+
+// Encryptor in an interface for encryptors
+//
+// encrypt with pk : ciphertext = [pk[0]*u + m + e_0, pk[1]*u + e_1]
+// encrypt with sk : ciphertext = [-a*sk + m + e, a]
+type Encryptor interface {
+	// EncryptNew encrypts the input plaintext using the stored key and returns
+	// the result on a newly created ciphertext. The encryption is done by first
+	// encrypting zero in QP, dividing by P and then adding the plaintext.
+	// The level of the output ciphertext is plaintext.Level().
+	EncryptNew(plaintext *Plaintext) *Element
+
+	// EncryptNTTNew encrypts the input plaintext using the stored key and returns
+	// the result on a newly created ciphertext. The encryption is done by first
+	// encrypting zero in QP, dividing by P and then adding the plaintext.
+	// The level of the output ciphertext is plaintext.Level().
+	// Output will be in the NTT domain.
+	EncryptNTTNew(plaintext *Plaintext) *Element
```

Comment:
Same comment as for the Decryptor, if these NTT variants are for CKKS, I would try to support one type of encryption at rlwe level and add a final NTT at the ckks level (if doable and it has not a too big impact on perf of course).

Reply:
Same, the decryptor creates the plaintext, so it must be told what the output domain is.

Reply:
Remove [...]New methods from RLWE and wrap inplace method in schemes

Reply:
Done